### PR TITLE
JENKINS-42006# Fix to return only authenticated user's permission

### PR DESCRIPTION
**Pull request by vivek**
# Description

See [JENKINS-42006](https://issues.jenkins-ci.org/browse/JENKINS-42006).

# Submitter checklist
- [x] Link to JIRA ticket in description, if appropriate.
- [x] Change is code complete and matches issue description
- [x] Appropriate unit or acceptance tests or explanation to why this change has no tests
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.
- [ ] Ran Acceptance Test Harness against PR changes.

# Reviewer checklist
- [ ] Run the changes and verified the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

@reviewbybees

### CODE REVIEW

**Comment by vivek** [comment]
@michaelneale there was one more fix needed for this.

File: blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/UserImpl.java [hunk 1]
```diff
@@ -88,8 +88,16 @@ public BlueUserPermission getPermission() {
...
+        User loggedInUser = User.get(name, false, Collections.EMPTY_MAP);
+        if(loggedInUser == null){
+            return null;
+        }
+
+        // If this user is not logged in, we do not show it's permissions
+        // XXX: This is done to avoid impersonation which has performance
+        //      implications, e.g. github oauth plugin might do a network
+        //      round trip to fetch user and authorizations
+        if(!loggedInUser.getId().equals(user.getId())){
```
Comment (michaelneale) [inline blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/UserImpl.java]: does that mean it does the overhead when logged in?

**Comment by michaelneale** [comment]
:bee:

File: blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/UserImpl.java (hunk 1 above)
Comment (vivek) [inline blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/rest/UserImpl.java]: no it doesn't. It just means for other users no permissions will be returned.

**Pull request merged**